core/commonMain/src/kotlinx/serialization/Serializers.kt

```diff
@@ -244,8 +274,8 @@ private fun KClass<Any>.compiledParametrizedSerializer(serializers: List<KSerial
 
 @OptIn(ExperimentalSerializationApi::class)
 private fun KClass<Any>.builtinParametrizedSerializer(
-    typeArguments: List<KType>,
     serializers: List<KSerializer<Any?>>,
+    classifierProvider: (Int) -> KClassifier?
```

Comment:
Given that this int is always `0`, maybe rename this parameter to `elementClassifierIfArray` and remove Int lambda parameter?